### ISSUE TITLE
Incorrect parsing related to `__` usage

### ISSUE THREAD

**Issue by treyhunner** [rvben/rumdl#17]
When linting this `file1.md` file:

```
Instead of assigning to `self.value`, we're relying on the [`__dict__`][__dict__] in our object to hold that value instead.

Hint:

- [An article on something](https://blog.yuo.be/2018/08/16/__init_subclass__-a-simpler-way-to-implement-class-registries-in-python/ "Some details on using `__init_subclass__`")


[__dict__]: https://www.pythonmorsels.com/where-are-attributes-stored/
```

All checks pass (except 3 which are disabled):

```bash
uv tool run rumdl check --disable MD012,MD013,MD041 file1.md

Success: No issues found in 1 file (26ms)
```

But this very similar file, `file2.md`:

```
We're doing this because generator functions return a generator object which [is an iterator][generators are iterators] and **we need `__iter__` to return an [iterator][]**.

Instead of assigning to `self.value`, we're relying on the [`__dict__`][__dict__] in our object to hold that value instead.

Hint:

- [An article on something](https://blog.yuo.be/2018/08/16/__init_subclass__-a-simpler-way-to-implement-class-registries-in-python/ "Some details on using `__init_subclass__`")


[__dict__]: https://www.pythonmorsels.com/where-are-attributes-stored/
[generators are iterators]: https://www.pythonmorsels.com/generators-are-iterators/
[iterator]: https://www.pythonmorsels.com/what-is-an-iterator/
```

Fails these same checks:

```bash
$ uv tool run rumdl check --disable MD012,MD013,MD041 file2.md
file2.md:3:73: [MD050] Strong emphasis should use ** instead of __ [*]
file2.md:7:60: [MD050] Strong emphasis should use ** instead of __ [*]
file2.md:10:2: [MD050] Strong emphasis should use ** instead of __ [*]

Issues: Found 3 issues in 1 file (32ms)
Run with `--fix` to automatically fix 3 of the 3 issues
```

The issue of `__` being used instead of `**` seems to be related to the `__` usage within either the link anchors (given the line numbers in the output) but it doesn't show up as an issue without the correct `**` bolding used on line 1.

**Comment by treyhunner** [comment]
I realized the `file1.md` and `file2.md` distinction was a bit of a red herring.

I added this to my `rumdl.toml`:

```
[MD050]
style = "asterisk"
```

Which then caused `file1.md` to also show errors:

```
$ uv tool run rumdl check --disable MD012,MD013,MD041 file1.md
file1.md:1:73: [MD050] Strong emphasis should use ** instead of __ [*]
file1.md:5:60: [MD050] Strong emphasis should use ** instead of __ [*]
file1.md:8:2: [MD050] Strong emphasis should use ** instead of __ [*]

Issues: Found 3 issues in 1 file (21ms)
Run with `--fix` to automatically fix 3 of the 3 issues
```

So the issue is that `__` usage within link anchors and URLs is being incorrectly interpreted as bolding.

**Comment by treyhunner** [comment]
I am not sure whether this is related or not, but this `file6.md`:

```
This works particularly well if all of the objects we're working with are **either truthy or `None`**.
```

Results in an odd linting error:

```
$ uv tool run rumdl check --disable MD041,MD013 file6.md
file6.md:1:75: [MD037] Spaces inside emphasis markers: "**either truthy or       **" [*]

Issues: Found 1 issues in 1 file (23ms)
Run with `--fix` to automatically fix 1 of the 1 issues
```

When the backticks are removed around `None`, the error goes away.

**Comment by sco1** [comment]
Also seeing similar issue with inline code:

```markdown
# Sample Markdown

This is an example line that talks about `__init__` methods.
```

Which yields:

```
$ rumdl --version
rumdl 0.0.106
$ rumdl check .
.\foo.md:3:43: [MD050] Strong emphasis should use ** instead of __ [*]
```
